emcc.py

```diff
@@ -649,6 +649,19 @@ def check(input_file):
 
 
 def process_dynamic_libs(dylibs):
+  extras = []
+  seen = set()
+  for dylib in dylibs:
+    dylink = webassembly.parse_dylink_section(dylib)[0]
+    for needed in dylink.needed:
+      if needed in seen:
+        continue
+      if os.path.exists(needed):
+        extras.append(needed)
+      else:
+        diagnostics.warning('emcc', '{dylib}: shared library dependency not found: `{needed}`')
```

Comment:
Should this be an f-string?

Reply:
(and, should this be an error at some point?)

Reply:
Add transitive dependency scanning and library path searching.. and made this into a hard error.